Clean-up for some packages in `haskell-modules/configuration-common.nix`.

Anton-Latukha opened the pull request:
###### Motivation for this change

Clean-ups and more informative docs to: `{haddock-library, haddock-library_1_9_0, pandoc (1.2.7), hnix, generic-lens, dhall, dhall-json, dhall-nix}`

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Anton-Latukha committed:
hnix: rm no longer relevant comments

Cleaning up old comments that do not hold.

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
dhall: document proposed idea upstream to solve `dontCheck`

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
dhall-json: rm upstream issue link, no loner relevant

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
dhall-json: enable check

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
haddock-library: enable deps and checks, they are fine

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
haddock-library_1_9_0: enable deps and checks, they are fine

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
haddock-library{,_1_9_0}: rm no longer relevant note

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
generic-lens: change upstream report link to relevant

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
pandoc: update the note why `dontCheck`

M  pkgs/development/haskell-modules/configuration-common.nix

cdepillabout commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -219,16 +219,7 @@ self: super: {
-    generateOptparseApplicativeCompletion "hnix" (
-      dontCheck (doJailbreak (super.hnix.override { these = self.these_0_7_6; }))
-    );
+  hnix = dontCheck (super.hnix);
```
Could you do two things here:

- Remove `hnix` from the broken packages section in the `pkgs/development/haskell-modules/configuration-hackage2nix.yaml`, since it is building now.
- Add a comment here why `dontCheck` is needed (if it makes sense).

Anton-Latukha replied:
Yep, `hnix` can not pass its testing suite.

And btw, maybe would add that  `gen*Optparse*Completion` plainly does not work - it falls the build.

I thought I've seen it is broken, but on yesterday's search, I did not find that, thanks for pointing it out. I am still learning infrastructure.

Anton-Latukha replied:
What was added:
  * `hnix` doc why `dontCheck`.
  * `hnix` doc why no `gen*Optparse*Completion`
  * rm `these ==0.7.6` from `configuration-hackage2nix.yaml` and overrides for it in `configuration-common.nix `, since this package version was used only by old `hnix` version

cdepillabout replied:
> And btw, maybe would add that gen*Optparse*Completion plainly does not work - it falls the build.

I tried 

```diff
diff --git a/pkgs/development/haskell-modules/configuration-common.nix b/pkgs/development/haskell-modules/configuration-common.nix
index 8409ff2778e..72f6d951ada 100644
--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -219,7 +219,7 @@ self: super: {
   # 2020-06-05: HACK: `generateOptparseApplicativeCompletion` does not work,
   # build fails - option disabled
   # 2020-06-05: HACK: does not passes own build suite - `dontCheck`
-  hnix = dontCheck (super.hnix);
+  hnix = generateOptparseApplicativeCompletion "hnix" (dontCheck (super.hnix));
 
   # Fails for non-obvious reasons while attempting to use doctest.
   search = dontCheck super.search;
```

and it appears to build.

What sort of error did you get when you tried this?

Anton-Latukha replied:
There was an error during the generation of the completion.

Maybe it is because I started with this package first, so before its clean-up, and as so package used for example `these` old vertion.

Anton-Latukha replied:
And indeed, the current upstream code fails to generate complettions:

```
Test suite logged to: dist/test/hnix-0.8.0-hnix-tests.log
1 of 1 test suites (1 of 1 test cases) passed.
haddockPhase
installing
Installing library in /nix/store/i5rk2nl0yaa2w4rbzqxmlv53jrambzyg-hnix-0.8.0/lib/ghc-8.8.2/x86_64-linux-ghc-8.8.2/hnix-0.8.0-Je1zkJQRPx69EBuUzc9dwC
/nix/store/ba8fhbmvfyh439606lni91y420yflmlc-stdenv-linux/setup: line 85: /nix/store/i5rk2nl0yaa2w4rbzqxmlv53jrambzyg-hnix-0.8.0/bin/hnix-0.8.0: No such file or directory
builder for '/nix/store/34kid81jvakazykkfsv7p8b2qpgx0b9i-hnix-0.8.0.drv' failed with exit code 127
error: build of '/nix/store/34kid81jvakazykkfsv7p8b2qpgx0b9i-hnix-0.8.0.drv' failed
```

So I would recommend for now leave it as is, until it would be figured-out.

Anton-Latukha replied:
Oh, I figured it out. `developPackage.name` is not a `package.name`, it adds a version to it.

Anton-Latukha replied:
Enabled completions.

Anton-Latukha replied:
I would be glad when completions are tested in the upstream CI.

cdepillabout reviewed (changes requested):
Other than the request for `hnix` below, this looks good.  I'll merge this in when the `hnix` things have been addressed.

Thanks for taking them time to fix this sort of stuff up!

Anton-Latukha committed:
hnix: rm from broken-packages

M  pkgs/development/haskell-modules/configuration-hackage2nix.yaml

Anton-Latukha committed:
hnix: add documentation why `dontCheck`

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
hnix: rm `these` override, package requires `these >= 1.0.2`

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
rm these_0_7_6, no longer used package version

Was require by old hnix version.

M  pkgs/development/haskell-modules/configuration-hackage2nix.yaml

Anton-Latukha committed:
hnix: rm `doJailbreak`

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha commented:
> Other than the request for `hnix` below, this looks good. I'll merge this in when the `hnix` things have been addressed.
> 
> Thanks for taking them time to fix this sort of stuff up!

Very good honest and effective responce to contributers, thank you 8)
